api/role-create/index.ts

```diff
@@ -0,0 +1,55 @@
+import "../startup";
+import { Context, HttpRequest } from "@azure/functions";
+import { authorized, ApiRequestContext } from "../common/ApiRequestContext";
+import * as Validator from "validatorjs";
+import { badRequest, badRequestFor, ok } from "../common/http/CommonResults";
+import { RoleService } from "../common/services/RoleService";
+
+export default async function (context: Context, req: HttpRequest): Promise<void> {
+  await authorized(
+    context,
+    req,
+    "admin",
+    async () => {
+      const data = { ...req.body } as RoleCreateForm
+      const validation = new Validator(data, roleCreateFormRules);
+
+      if (validation.fails()) {
+        context.res = badRequest(validation);
+        return;
+      }
+
+      const roleService = new RoleService();
+      let { exists } = await roleService.getRoleByName(data.name);
+    
+      if (exists) {
+        context.res = badRequestFor({
+          name: ["This name is not available."]
+        });
+        return;
+      }
+
+      const role = await roleService.createRole(data);
+      if (!role) {
+        context.res = {
+          status: 500,
+          body: JSON.stringify({ success: false, reason: "Internal error creating role" })
+        };
+        return;
+      }
+
+      context.res = ok("created");
+    },
+    true
```

Comment:
This true causes the currently logged in users data to be loaded from cosmosdb - the loaded user data isn't used anywhere in this function so it could be false / omitted (it defaults to false).